internal/cli/openapi.go

```diff
@@ -78,7 +84,48 @@ func (c *openapiCmd) run(ctx context.Context, args []string) error {
 		return err
 	}
 
-	encoder := json.NewEncoder(os.Stdout)
+	encoder := json.NewEncoder(c.streams.Out)
 	encoder.SetIndent("", "  ")
 	return encoder.Encode(msg)
 }
+
+type headlessServer struct {
+	memconn              apiserver.ConnProvider
+	hudsc                *server.HeadsUpServerController
+	loopbackClientConfig *rest.Config
+}
+
+func newHeadlessServer(ctx context.Context) (*headlessServer, error) {
+	tmpdir, err := ioutil.TempDir("", "tilt-headless")
+	if err != nil {
+		return nil, err
+	}
+	dir := dirs.NewTiltDevDirAt(tmpdir)
```

Comment:
Not a huge deal as-is (and I can very much see the argument against adding complexity here), but we could change `HeadsUpServerController::addToAPIServerConfig` -> `maybeAddToAPIServerConfig` (same with remove) and only do work if `configAccess != nil` and then everything would work here w/o needing to generate temporary files.

Reply:
ooh that's clever! done